Around 1 in 17 people is affected by one of ~7,000 known rare diseases. Most of these patients do not receive a diagnosis, which means they remain in uncertainty without a prognosis, are unable join specific patient support groups, and do not receive the most appropriate treatment.
Next-generation sequencing (NGS) of DNA promises to establish a molecular diagnosis and help these patients but many challenges still stand in the way of maximum success.
Recent years have seen great advances in computational tools that quickly reduce the amount of DNA variants to be interpreted by a human expert for potentially pathogenic effects.
But, the current tools that rely on features such as evolutionary conservation, annotation of regulatory genomics elements and structural DNA features have been already optimized over many years and significant improvements are not expected. 
Here we tried to introduced structural features of proteins into diagnostics based on the methods used by VIPUR. Through the difficulties of protein modeling and experts knowledge it was discovered that methods used by VIPUR are not features that can help in diagnosis with machine learning. Structural data of proteins is often incomplete and is highly dependent on experimentally determined structures which are expensive to make. The methods that VIPUR uses to standardize protein structures for machine learning removes the context and treats them like they are in a vacuum. To gain a more realistic view on structural information we chose to use the web service HOPE. We also developed a method to gain insight in the structural features of a protein and several of its variants. With these methods we tried to collect structural information which is usable for diagnosis.