%INTRO
People with rare diseases are currently hard to diagnose and are often not put in the desired treatment groups, with machine learning methods such as GAVIN 95\% of the benign variants can be harmfully removed, however these methods rely on conservation and have been heavily optimized over the years \cite{van_der_velde_gavin:_2017}.
%GAVIN: Gene-Aware Variant INterpretation for medical sequencing, van der Velde
A new refreshing approach called VIPUR uses sequential and structural data to predict deleteriousness of a protein variants.

%VIPUR
Within the attempt to make VIPUR usable for the diagnosis of rare diseases it was discovered that some questionable steps were taken to make it especially applicable for diagnosis but also to determine deleteriousness of proteins itself: (i) "All protein models were standardized to remove unwanted components (duplicate chains, ligands, metals and non-standard amino acids)" \cite{baugh_robust_2016}.
%Robust classification of protein variation using structural modelling and large-scale data integration, Baugh
Standardizing data can be beneficial to avoid learning features from proteins that are available to some models but should not be the determining factor for classification. However any form of context to the protein is removed and might therefore make incorrect assumptions about how: a monomer interacts with other monomers, ligands, metals, non-standard amino acids and water which can all have an effect on how proteins shape and interact \cite{koshland_application_1958}.
%Application of a Theory of Enzyme Specificity to Protein Synthesis*, Koshland
(ii) With the utilization of Rosetta's Relax application different models are formed based on the Monte Carlo method (Section \ref{section:Chap_Monte_Carlo}), VIPUR produces 50 structures with Relax per protein which is a tiny amount of the potential search space of possible folds that could have made changes in a mutated protein, which also can be seen in the scores of the model made from of TNFRSF1A with TNF $\alpha$ \& $\beta$ (Figures \ref{fig:relax_TNFA_scores}, \ref{fig:relax_TNFB_scores}), Rosetta itself suggests to make sufficient models, starting with 5000\cite{rosetta_commons_analyzing_nodate}.
%Analyzing Results, Rosetta commons
(iii) The features acquired with probe in combination with the models that were produced, within the publication of Probe is mentioned: "It requires both highly accurate structures and also the explicit inclusion of all hydrogen atoms and their van der Waals interactions." \cite{word_visualizing_1999}.
%Visualizing and quantifying molecular goodness-of-fit: small-probe contact dots with explicit hydrogen atoms11Edited by J. Thornton, Word et al.
It is currently not possible to determine if the structures were accurate, however is it likely that no loose hydrogen atoms were included within the structure based on the knowledge that all structures were standardized and likely some of the structures did not have any loose hydrogens within them. To make the outcome of Probe useful to VIPUR the program Reduce should have ran first, which add hydrogen atoms to the structure, which is recommend on the site were Probe can be downloaded from \cite{richardson_lab_probe_nodate}.
%Probe Software : Kinemage Website, Richardson LAB

More questions arise when further investigating the publication; within the figures (4, 5) \cite{baugh_robust_2016} and supplementary figures (10, 11) \cite{baugh_supplementary:_2016} are heatmaps of PSSMs added that display the values of the natural and mutated residues. In combination with the methods used on standardizing structures and collecting features a suspicion arises that there is little contribution from the structural features and that prediction depends on PSI-BLAST results. Figure 4 within the publication shows a protein wherein residue serine 204 which part of an $\alpha$ helix is mutated to proline and is predicted as deleteriouness by VIPUR, which is logical even without predictions because prolines are known to be $\alpha$ helix stoppers \cite{li_alpha-helical_1996} and therefor affect the $\alpha$ helix it form.
%Robust classification of protein variation using structural modelling and large-scale data integration, Baugh
%SUPPLEMENTARY: Robust classification of protein variation using structural modelling and large-scale data integration, Baugh et al.
%Alpha-helical, but not beta-sheet, propensity of proline is determined by peptide environment, Li et al.

% Your VIPUR usage
VIPUR has not been used for various reasons; The models that were predicted by the web services (Figures \ref{fig:I_Tasser_Robetta_models}) had a decent accuracy for the binding site, but the rest of structures differ in various sections and make it hard to determine if these were accurate representations of TNFRSF1A; VIPUR was built on and for a single system and required reverse engineering to test the basic demos and PyMOL or PyRosetta (Sections \ref{subsec:MM_PyMOL}, \ref{subsec:MM_PyRosetta}) still had to be replaced with modeller (Section \ref{subsec:MM_Modeller}) which would have required more time to repair. 
The differences in between the models are probably due to the fact that TNFRSF1A is transmembrane protein which is hard to acquire structures from with experimental methods \cite{yonath_x-ray_2011}, even though $\sim57\%$ of TNFRSF1As structure was known (1EXT\cite{naismith_structures_1996} , 1ICH\cite{sukits_solution_2001}), if more models would have been produced a potential accurate structure could have been formed .
%Structures of the extracellular domain of the type I tumor necrosis factor receptor, Naismith
%Solution structure of the tumor necrosis factor receptor-1 death domain, Suktis et al.

% last part of VIPUR
Although it is not part of developing a technique that can help diagnosing rare disease variants but the publication contains a claim ("VIPUR can be applied to mutations in any organismâ€™s proteome...." \cite{baugh_robust_2016}) which contradicts with its methods: "remove unwanted components (duplicate chains, ligands, metals and non-standard amino acids)" \cite{baugh_robust_2016}. Currently there are more than 140 amino acids found in natural proteins of which 22 are part of the amino acid alphabet and 20 of those are classified as standard \cite{ambrogelly_natural_2007}. By removing the non-standard amino acids from proteins it becomes impossible to analyze mutations in every organisms proteome.
%Robust classification of protein variation using structural modelling and large-scale data integration, Baugh
%Natural expansion of the genetic code, Ambrogelly et al.

% SPVAA
SPVAA did not assess whole complex and neither did became a machine learning tool ready to use in diagnosis to make automated predictions. The protein information have similar weaknesses as VIPUR wherein: water, metals and other molecules are removed from its structure. However the proteins can keep their extra monomers and protein ligands, even when the structure requires identical or different ones they can be added manually into the structure and analyzed. 

For three variants it has been attempted to acquire structural information to determine their differences with the wild type structure, in two of the three structures hardly any changes were visible. From the models that were assessed only homotrimeric TNF$\alpha$ -$\beta$ bound mutations were processed, not the unboud dimeric structures. From the proteins that were modeled too few were produced of each structure, backrub used 10000 Monte Carlo moves, which is very little compared to the amount of residues it had and should have required more. The relax application made 64 models per mutation but Rosetta itself suggest to make at least 5000\cite{rosetta_commons_analyzing_nodate}.
%Analyzing Results, Rosetta commons
The only models with mutations where pathogencity was highly likely visible were the CYS 62 GLY models that had broken disulfide bridges that could lead to instability in the protein, based on the scores produced by modeller little could be discovered except the higher scores in the disulfide geometry potential (dslf\_fa13) of CYS 62 GLY. Many of the scores show overlap and are difficult to relate with the applied structural changes, it could have been that many of the distributions should overlap more but because few models were made they tend to separate.

The mutated models could have been produced more accurate in the context of making potential disulfide bridges, with the current program they are guessed based on distance which could lead to the insertion of too many or few disulfide bridges. A better method which modeller has to form disulfide bridges based on original data that is available within structures. 

% MOLECULAR DYNAMICS
VIPUR and SPVAA could both be improved in various ways, one of them would be by doing molecular dynamic simulations on the mutated structures to determine the effects of structural changes. With SPVAA it would have most likely become clear if the loss of the disulfide bridge from CYS 62 GLY  in TNFRSF1A would have caused structural issues. VIPUR could benefit from it as new machine learning feature in situations where limited movement is observed and it changes tremendously when a missense mutation occurred in a protein increases with a mutation or vice versa.

%HOPE
HOPE is informative with the results it produces, easy to use, fast and makes structural problems within proteins diagnoseable and understandable when a missense mutation is discovered. However it does not draws a solid conclusion and the information it collects depends on: previous publications, conservation and experimental structures, which will give it a disadvantage when limited knowledge is available. Also it does not asses a complex but it can describe binding sites from the monomer when previously discovered. For the mutation CYS 62 GLY was very clear based on conservation, however for the pathogenic mutation PHE 141 ILE the information was less clear. The provisional likely benign glutamic acid at position 138 makes salt bridges according to HOPE which are strong bonds and can be important for internal structures and binding. With this information GLU 138 ALA would be likely pathogenic, however change in interaction or structure have not been observed with HOPE or SPVAA and therefore can not be classified pathogenic with certainty.


