
People with rare diseases are hard to diagnose

Prediction of pathogenicity in variants momentarily done based on sequence information and has been successful for certain groups of genes \cite{}.
%GAVIN: Gene-Aware Variant INterpretation for medical sequencing, van der Velde
However pathogenticy of some genes with their variants cannot be classified by the currently used features for classification. Recently a method, called VIPUR, surfaced that incorporated sequence and protein data for classification of the pathogenicity from gene variants\cite{}. 
%Robust classification of protein variation using structural modelling and large-scale data integration, Baugh

In the attempt to reproduce the methods taken by the VIPUR approach on protein structures that are related to rare diseases it was realized that some questionable steps were being taken. With this approach all ligands were removed \cite{} which changes the energies within and can therefor alter the structure \cite{} and causes it to be analyzed from a single perspective instead of two when a bound ligand is also taken into account. 
%Robust classification of protein variation using structural modelling and large-scale data integration, Baugh
%Application of a Theory of Enzyme Specificity to Protein Synthesis*, Koshland
Another step that was taken with VIPUR is that each structure is viewed as a monomer which is for some proteins not a problem, but for a complex that consists of multiple similar or a variety of different monomers makes it difficult to assess the effects.

To make predictions for new benign and pathogenic variants from TNFRSF1A, more information should be collected on how certain residues contribute to TNFRSF1A. More differences between interaction energies in mutated proteins could have been found be by adding molecular dynamics (MD) simulations of TNF $\alpha$/$\beta$ separately TNFRSF1A and combined with TNF docked into TNFRSF1A.



 prediction of potential benign and pathogenic variants of TNFRSF1A isoforms should be included in the analysis to gain insight in which part of the proteins are highly important for the interactions and could result in a better prediction.

 A significant contribution to gain more insight in how TNFRSF1A interacts with TNF $\alpha$/$\beta$ would have been the addition of molecular dynamic simulations; it shows how the proteins move on their own but also how the residues of the protein and the ligand interact with each other.



Isoforms were not taken into account.



 some steps have become questionable in structure of TNFRSF1A some questionable  rare diseases  some questionable  training set some q to reproduce some of the results that were acquired with the VIPUR some questionable assu


Looking at the investigation t

With the resource at our disposal we were unable to reproduce any of the results that were produced by VIPUR for testing purposes, by 