
People with rare diseases are hard to diagnose

Prediction of pathogenicity in variants momentarily done based on sequence information and has been successful for certain groups of genes \cite{}.
%GAVIN: Gene-Aware Variant INterpretation for medical sequencing, van der Velde
However pathogenticy of some genes with their variants cannot be classified by the currently used features for classification. Recently a method, called VIPUR, surfaced that incorporated sequence and protein data for classification of the pathogenicity from gene variants\cite{}. 
%Robust classification of protein variation using structural modelling and large-scale data integration, Baugh

In the attempt to reproduce the methods taken by the VIPUR approach on protein structures that are related to rare diseases it was realized that some questionable steps were being taken. With this approach all ligands were removed \cite{} which changes the energies within and can therefor alter the structure \cite{} and causes it to be analyzed from a single perspective instead of two when a bound ligand is also taken into account. 
%Robust classification of protein variation using structural modelling and large-scale data integration, Baugh
%Application of a Theory of Enzyme Specificity to Protein Synthesis*, Koshland
Another step that was taken with VIPUR is that each structure is viewed as a monomer which is for some proteins not a problem, but for a complex that consists of multiple similar or a variety of different monomers makes it difficult to assess the effects.

To make predictions for new benign and pathogenic variants from TNFRSF1A, more information should be collected on how certain residues contribute to TNFRSF1A. More differences between interaction energies in mutated proteins could have been found be by adding molecular dynamics (MD) simulations of TNF $\alpha$/$\beta$ separately TNFRSF1A and combined with TNF docked into TNFRSF1A.

We did not asses the whole in a complex because there was insufficient information. (Membrane, water, other atoms and we could have left some atoms that were not water.)
All methods are reliant on existing structures making it difficult to asses it as a complex.


 prediction of potential benign and pathogenic variants of TNFRSF1A isoforms should be included in the analysis to gain insight in which part of the proteins are highly important for the interactions and could result in a better prediction.

 A significant contribution to gain more insight in how TNFRSF1A interacts with TNF $\alpha$/$\beta$ would have been the addition of molecular dynamic simulations; it shows how the proteins move on their own but also how the residues of the protein and the ligand interact with each other.
 
 Not certain of how VIPUR used Modeller but basing it on the removal of metals and water during SPVAA it likely also removed non-standard amino acids and therefore making the method unfit to predict every type of protein made as claimed.

Only using the Rosetta energy to determine the effect of the protein.

The VIPUR pipeline could not be executed because it was not possible to compile PyRosetta or PyMOL on the cluster.

however VIPUR has not been tested due to not having the correct software available and TNFRSF1A was not within the training data set of VIPUR. 

Isoforms were not taken into account.

Within the publication of Probe is mentioned: "It requires both highly accurate structures and also the explicit inclusion of all hydrogen atoms and their van der Waals interactions." \cite{}.
%Visualizing and quantifying molecular goodness-of-fit: small-probe contact dots with explicit hydrogen atoms11Edited by J. Thornton, Word

The site of Probe mentions: "Meaningful analysis of molecular contact surfaces requires that ALL atoms are considered. Before using Probe, use the companion program Reduce to add hydrogens to the coordinate file." \cite{}, no mention of the Reduce software is mentioned in the VIPUR approach and therefor it is difficult to asses the meaning of previously acquired results in the VTS.
%Probe Software : Kinemage Website, Richardson LAB

VIPUR is questionable because it has a limited amount of simulations.
VIPUR uses PSI-blast to justify its results.

Describe the error made with apply mutations from different isoforms, see the table within results.

Only assessed a small piece of TNFRSF1A and did not even look at the class of proteins itself.

More iterations on Rosetta relax than 64\cite{}.
%Analyzing Results, Rosetta commons




Good other suggestions for finding if the approach really means something is by using shap \cite{}.
%slundberg/shap: A unified approach to explain the output of any machine learning model, Slundberg
%Explaining Prediction Models and Individual Predictions with Feature Contributions, Å trumbelj
%Why Should I Trust You?": Explaining the Predictions of Any Classifier, Ribeiro
%Learning Important Features Through Propagating Activation Differences, Shrikumar
%Algorithmic Transparency via Quantitative Input Influence: Theory and Experiments with Learning Systems, Datta
%On Pixel-Wise Explanations for Non-Linear Classifier Decisions by Layer-Wise Relevance Propagation, Bach
%Interpreting random forests | Diving into data, Datadive

The software written has a limited use currently and could be expanded to rapidly introduce mutations in multiple structures and chains at once.

It might have been useful to disable disulfide bridges if no cysteine residue is mutated because it is less likely that alterations are formed to disulfide bridges and otherwise they might be added without a reason.

Fragments were only available of 1EXT and 1TNR because it is a transmembrane protein which is difficult to make a structure from with X-ray crystallography.
Maybe the first 30 residues were unnecessary because the might be signal peptides. 

A good thing from VIPUR is that they took the information of all samples, it might introduce errors but also excludes lowers the extremes and might make it more realistic.

 some steps have become questionable in structure of TNFRSF1A some questionable  rare diseases  some questionable  training set some q to reproduce some of the results that were acquired with the VIPUR some questionable assu

With SPVAA only looked at the trimeric form an not the dimeric form of TNFRSF1A.


Looking at the investigation t

With the resource at our disposal we were unable to reproduce any of the results that were produced by VIPUR for testing purposes, by 