%INTRO
Currently, most rare disease patients do not receive a molecular diagnosis.
Despite machine learning methods such as GAVIN that can remove 95\% of benign variation from the genome\cite{van_der_velde_gavin:_2017}, it is still very difficult to pin-point causal variants in the genome.
Such methods rely mostly on evolutionary conservation and have been heavily optimized over the years.
Therefore we need new refreshing approaches such as VIPUR, which uses sequential and structural data instead, that have the exciting potential to help us diagnose more patients. 
%GAVIN: Gene-Aware Variant INterpretation for medical sequencing, van der Velde

\subsection{VIPUR}
%VIPUR
Within the attempt to make VIPUR usable for diagnostics it was discovered that some questionable steps were taken to make it applicable for diagnosis but also to determine deleteriousness ; (i) "All protein models were standardized to remove unwanted components (duplicate chains, ligands, metals and non-standard amino acids)" \cite{baugh_robust_2016}.
%Robust classification of protein variation using structural modelling and large-scale data integration, Baugh
Standardizing data can be beneficial to avoid learning features from proteins that are available to some models but should not be the determining factor for classification. However any form of context to the protein is removed and might therefore make incorrect assumptions about how: a monomer interacts with other monomers, ligands, metals, non-standard amino acids and water which can all have an effect on how proteins shape and interact \cite{koshland_application_1958}.
%Application of a Theory of Enzyme Specificity to Protein Synthesis*, Koshland
(ii) With the utilization of Rosetta's Relax application different models are formed based on the Monte Carlo method (Section \ref{section:Chap_Monte_Carlo}). VIPUR produces 50 structures with Relax per protein which is a tiny amount of the potential search space of possible folds that could have made changes in a mutated protein, which is also visible in the scores of the models made from of TNFRSF1A with TNF $\alpha$ \& $\beta$ (Figures \ref{fig:relax_TNFA_scores}, \ref{fig:relax_TNFB_scores}). Rosetta itself suggests to make sufficient models, starting with a minimum of 5000\cite{rosetta_commons_analyzing_nodate}.
%Analyzing Results, Rosetta commons
(iii) The features acquired with probe in combination with the models that were produced, within the publication of Probe is mentioned: "It requires both highly accurate structures and also the explicit inclusion of all hydrogen atoms and their van der Waals interactions." \cite{word_visualizing_1999}.
%Visualizing and quantifying molecular goodness-of-fit: small-probe contact dots with explicit hydrogen atoms11Edited by J. Thornton, Word et al.
It is not possible to determine if the structures were accurate. However is it likely that no loose hydrogen atoms were included within the structures because all were standardized. To make the outcome of Probe useful to VIPUR the program Reduce should have been executed first, which adds hydrogen atoms to the structure. It is recommended on the site of Probe to execute Reduce on the structure before using VIPUR \cite{richardson_lab_probe_nodate}.
%Probe Software : Kinemage Website, Richardson LAB

More questions arise when further investigating the publication. Within the figures (4, 5) \cite{baugh_robust_2016} and supplementary figures (10, 11) \cite{baugh_supplementary:_2016} are heatmaps of PSSMs added that display the values of the natural and mutated residues. In combination with the methods used on standardizing structures and collecting features a suspicion arises that there is little contribution from the structural features and that prediction depends on PSI-BLAST results. Figure 4 within the publication shows a protein wherein serine 204 which part of an $\alpha$ helix is mutated to proline and is predicted as deleteriouness by VIPUR. Which is logical even without predictions because prolines are known to be $\alpha$ helix stoppers \cite{li_alpha-helical_1996} and therefor affect the form of an $\alpha$ helix (which is also not visible in the figure.).
%Robust classification of protein variation using structural modelling and large-scale data integration, Baugh
%SUPPLEMENTARY: Robust classification of protein variation using structural modelling and large-scale data integration, Baugh et al.
%Alpha-helical, but not beta-sheet, propensity of proline is determined by peptide environment, Li et al.

% Your VIPUR usage
VIPUR has not been used for various reasons. The models that were predicted by the web services (Figures \ref{fig:I_Tasser_Robetta_models}) had a decent accuracy for the binding site, but the rest of structures differed and made it too difficult to determine which model was an accurate representation of TNFRSF1A. VIPUR was built on and for a single system and required reverse engineering to make it work in any form even without the substitution of PyMOL and PyRosetta  (Sections \ref{subsec:MM_PyMOL}, \ref{subsec:MM_PyRosetta}) with Modeller (Section \ref{subsec:MM_Modeller}).
The differences in models is probably due to the fact that TNFRSF1A is transmembrane protein which is hard to acquire structures from with experimental methods \cite{yonath_x-ray_2011}, even though $\sim57\%$ of TNFRSF1As structure was known (1EXT\cite{naismith_structures_1996} , 1ICH\cite{sukits_solution_2001}).
%Structures of the extracellular domain of the type I tumor necrosis factor receptor, Naismith
%Solution structure of the tumor necrosis factor receptor-1 death domain, Suktis et al.

% last part of VIPUR
Although it is not part of developing a technique that can help diagnosing rare disease variants but the publication contains a claim ("VIPUR can be applied to mutations in any organismâ€™s proteome...." \cite{baugh_robust_2016}) which contradicts with its methods: "remove unwanted components (duplicate chains, ligands, metals and non-standard amino acids)" \cite{baugh_robust_2016}. Currently there are more than 140 amino acids found in natural proteins of which 22 are part of the amino acid alphabet and 20 of those are classified as standard \cite{ambrogelly_natural_2007}. By removing the non-standard amino acids from proteins it becomes impossible to analyze mutations in every organisms proteome.
%Robust classification of protein variation using structural modelling and large-scale data integration, Baugh
%Natural expansion of the genetic code, Ambrogelly et al.

\subsection{SPVAA}
% SPVAA
SPVAA did not assess whole complex and neither did became a machine learning tool ready to use in diagnosis to make automated predictions. SPVAA has similar weaknesses as VIPUR wherein: water, metals and other molecules are removed from its structure. However the proteins can keep their extra monomers and protein ligands, even when the structure requires identical or different ones they can be added manually into the structure and analyzed. 

For three variants it has been attempted to acquire structural information to determine their differences with the wild type structure, in two of the three structures hardly any changes were visible. From the models that were assessed only homotrimeric TNF$\alpha$ -$\beta$ bound mutations were processed, not the unboud dimeric structures. From the proteins that were modeled too few were produced of each structure. Backrub used 10000 Monte Carlo moves, which is very little compared for the amount of residues it has and should have had more to make larger changes in the structure. The relax application made 64 models per mutation but Rosetta itself suggest to make at least 5000\cite{rosetta_commons_analyzing_nodate}.
%Analyzing Results, Rosetta commons
The only models with mutations where pathogencity was highly likely visible were the CYS 62 GLY models, that had broken disulfide bridges that could lead to instability in the protein. Based on the scores produced by Relax little could be discovered except the higher scores in the disulfide geometry potential (dslf\_fa13) of CYS 62 GLY. Many of the scores show overlap and are difficult to relate with the applied structural changes. It could have been that many of the relax score distributions should have had more overlap with each because too few models were made.

The models could have had in other situations better disulfide bridge formation which did not rely on the guess of distance between cysteine residues. A better method which Modeller has is to form disulfide bridges based on other protein data that is available in some situations. Within TNFRSF1A it likely did not matter too much because except CYS 62 GLY all other methods show identical disulfide bridges. 

\subsection{HOPE}
%HOPE
HOPE is an informative tool that is easy to use, fast and makes structural problems within proteins understandable when a missense mutation is discovered. However it does not draws a solid conclusions and the information it collects depends on: previous publications, conservation and experimental structures. HOPE has a disadvantage when limited knowledge is available. Also it does not asses a complex but it can describe binding sites from the monomer when previously discovered. For the mutation CYS 62 GLY it was very clear based on conservation and the publication that it was pathogenic. However for the pathogenic mutation PHE 141 ILE the information was less clear. Glutamic acid at position 138 makes salt bridges according to HOPE which are in general strong bonds and can be important for internal structures and binding. With this information GLU 138 ALA would be likely pathogenic, however it does not draw a conclusion. Also no change in structure or interaction has been observed with HOPE or SPVAA and therefore can not be classified pathogenic with certainty and would therefor remain likely benign.
