%INTRO
People with rare diseases are currently hard to diagnose and are often not put in the desired treatment groups, with machine learning methods such as GAVIN 95\% of the benign variants can be harmfully removed, however these methods rely on conservation and have been heavily optimized over the years \cite{}.
%GAVIN: Gene-Aware Variant INterpretation for medical sequencing, van der Velde
A new refreshing approach called VIPUR uses sequential and structural data to predict deleteriousness of a protein variants.

%VIPUR
Within the attempt to make VIPUR usable for the diagnosis of rare diseases it was discovered that some questionable steps were taken to make it especially applicable for diagnosis but also to determine deleteriousness of proteins itself: (i) "All protein models were standardized to remove unwanted components (duplicate chains, ligands, metals and non-standard amino acids)" \cite{}.
%Robust classification of protein variation using structural modelling and large-scale data integration, Baugh
Standardizing data can be beneficial to avoid learning features that do not matter and can reduce overfitting. However any form of context to the protein is removed and might therefore make incorrect assumptions about how: a monomer interacts with other monomers, ligands, metals, non-standard amino acids and water which can all have an effect on how proteins shape and interact \cite{}.
%Application of a Theory of Enzyme Specificity to Protein Synthesis*, Koshland
(ii) With the utilization of Rosetta's Relax application different models are formed based on the Monte Carlo method (Section \ref{section:Chap_Monte_Carlo}), VIPUR produces 50 structures with Relax per protein which is a tiny amount of the potential search space of possible folds that could have made changes in a mutated protein, which also can be seen in the scores of the model made from of TNFRSF1A with TNF $\alpha$ \& $\beta$ (Figures \ref{fig:relax_TNFA_scores}, \ref{fig:relax_TNFB_scores}), Rosetta itself suggests to make sufficient models, starting with 5000\cite{}.
%Analyzing Results, Rosetta commons
(iii) The features acquired with probe in combination with the models that were produced, within the publication of Probe is mentioned: "It requires both highly accurate structures and also the explicit inclusion of all hydrogen atoms and their van der Waals interactions." \cite{}.
%Visualizing and quantifying molecular goodness-of-fit: small-probe contact dots with explicit hydrogen atoms11Edited by J. Thornton, Word et al.
It is currently not possible to determine if the structures were accurate, however is it likely that no loose hydrogen atoms were included within the structure based on the knowledge that all structures were standardized and likely some of the structures did not have any loose hydrogens within them. To make the outcome of Probe useful to VIPUR the program Reduce should have ran first, which add hydrogen atoms to the structure, which is recommend on the site were Probe can be downloaded from \cite{}.
%Probe Software : Kinemage Website, Richardson LAB
%The site of Probe mentions: "Meaningful analysis of molecular contact surfaces requires that ALL atoms are considered. Before using Probe, use the companion program Reduce to add hydrogens to the coordinate file." \cite{}, no mention of the Reduce software is mentioned in the VIPUR approach and therefor it is difficult to asses the meaning of previously acquired results in the VTS.

? easy pick ?
Within Figure 4 of Robust classification of protein variation using structural modelling and large-scale data integration \cite{} is stated that the mutation on position 204 from serine to proline is predicted deleteriousness, which is even without assessment highly likely because as can been in the structure it resides within an $\alpha$ helix and proline is known as the helix-stopper \cite{}.
%Robust classification of protein variation using structural modelling and large-scale data integration, Baugh
%Alpha-helical, but not beta-sheet, propensity of proline is determined by peptide environment, Li et al.

% Your VIPUR usage
VIPUR was not tested because of; several compilation issues related to libraries making not possible to compile PyMOL or PyRosetta (Sections \ref{subsec:MM_PyMOL}, \ref{subsec:MM_PyRosetta}), not able to run since it was made specifically for a single system and by modifying it unable to execute the basic implemented demo and due to the lack of time to reverse engineer Modeller (Section \ref{subsec:MM_Modeller}) into VIPUR. 

%Robust classification of protein variation using structural modelling and large-scale data integration, Baugh et al.
%SUPPLEMENTARY: Robust classification of protein variation using structural modelling and large-scale data integration, Baugh et al.

% Compilation issues
With the resource at our disposal we were unable to reproduce any of the results that were produced by VIPUR for testing purposes.
The VIPUR pipeline could not be executed because it was not possible to compile PyRosetta or PyMOL on the cluster.

%adquate thing about VIPUR quartiles
A good thing from VIPUR is that they took the information of all samples, it might introduce errors but also excludes lowers the extremes and might make it more realistic.


Based on the figures (4, 5) \cite{}, supplementary figures  (10, 11) \cite{} and methods on which structures and properties have been collected, there is a suspicion that there is little contribution from the structural properties in the machine learning method VIPUR and that most predictions are based on the results of PSI-BLAST.

% last part of VIPUR
Although it is not part of developing a technique that can help diagnosing rare disease variants but the publication contains a claim ("VIPUR can be applied to mutations in any organismâ€™s proteome...." \cite{}) which contradicts with its methods: "remove unwanted components (duplicate chains, ligands, metals and non-standard amino acids)" \cite{}. Currently there are more than 140 amino acids found in natural proteins of which 22 are part of the amino acid alphabet and 20 of those are classified as standard \cite{}. By removing the non-standard amino acids from proteins it becomes impossible to analyze mutations in every organisms proteome.
%Robust classification of protein variation using structural modelling and large-scale data integration, Baugh
%Natural expansion of the genetic code, Ambrogelly et al.



% SPVAA
SPVAA method show similar weaknesses as VIPUR, it removes metal and water but keeps the ligands and are even added when necessary.

We did not asses the whole in a complex because there was insufficient information. (Membrane, water, other atoms and we could have left some atoms that were not water.)

With SPVAA only looked at the trimeric form an not the dimeric form of TNFRSF1A.
Only using the Rosetta energy to determine the effect of the protein.

Isoforms were not taken into account.

Only assessed a small piece of TNFRSF1A and did not even look at the class of proteins itself.

With the resource at our disposal  limited.
More iterations on Rosetta relax than 64\cite{}.
%Analyzing Results, Rosetta commons

The software written has a limited use currently and could be expanded to rapidly introduce mutations in multiple structures and chains at once.

It might have been useful to disable disulfide bridges if no cysteine residue is mutated because it is less likely that alterations are formed to disulfide bridges and otherwise they might be added without a reason.

Fragments were only available of 1EXT and 1TNR because it is a transmembrane protein which is difficult to make a structure from with X-ray crystallography.
Maybe the first 30 residues were unnecessary because the might be signal peptides. 

%HOPE
HOPE is informative with the results it produces, easy to use, fast and makes structural problems within proteins diagnoseable and understandable when a missense mutation is discovered. However it does not draws a solid conclusion and the information it collects depends on: previous publications, conservation and experimental structures, which will give it a disadvantage when limited knowledge is available. Also it does not asses a complex but it can describe binding sites from the monomer when previously discovered.


% MOLECULAR DYNAMICS
SPVAA's assement and VIPURs prediction could potentially be improved with the addition of molecular dynamic simulations to determine the effects of structural changes. With SPVAA it would have most likely become clear if the loss of the disulfide bridge from cysteine 62 in TNFRSF1A would have caused structural issues. VIPUR could benefit from it as new machine learning feature in situations where limited movement is observed and it changes tremendously when a missense mutation occurred in a protein increases with a mutation or vice versa.



% General problem with all methods
All methods are reliant on existing structures making it difficult to asses it as a complex.
