\subsection{VIPUR approach}
Variant interpretation using Rosetta (VIPUR) is a machine learning approach for predicting deleteriousness of proteins (loss of function) and uses sequential and structural information. 
To train it the VIPUR training set (VTS) was made which contained sequential an structural features from protein structures that were acquired from Modbase \cite{pieper_modbase_2009} and
% modbase, a database of annotated comparative protein structure models and associated resources, pieper
SWISS-MODEL \cite{guex_automated_2009,bertoni_modeling_2017,benkert_toward_2011,bienert_swiss-model_2017,waterhouse_swiss-model:_2018}.
% Automated comparative protein structure modeling with SWISS-MODEL and Swiss-PdbViewer: A historical perspective, Guex
% Modeling protein quaternary structure of homo- and hetero-oligomers beyond binary interactions by homology, Bertoni
% Toward the estimation of the absolute quality of individual protein structure models, Benkert
% The SWISS-MODEL Repositoryâ€”new features and functionality, Bienert
% SWISS-MODEL: homology modelling of protein structures and complexes, Waterhouse
Proteins that did not have a structure in modbase or SWISS-MODEL were modeled with Modeller (Section \ref{subsec:MM_Modeller}) from proteins that had the highest amino acid sequence identity to the desired protein.
based on protein fragments that had the highest amino acid sequence identity to the protein
Some structures from the databases had: duplicate chains, ligands, metals and non-standard amino acids which were removed to avoid inconsistencies for the generated features by tools.
For all proteins a mutation file was made that described which and how many residues had to be mutated, with this file DDG monomer (Section \ref{subsubsec:MM_DDG_Monomer}) could determine changes in the protein structure, most results by ddg were used as features.
Structural mutations of proteins that are in the VTS were introduced by a script using PyMOL (Section \ref{subsec:MM_PyMOL}) by default or PyRosetta (Section \ref{subsec:MM_PyRosetta}) when PyMOL was not available.
After a mutation the structure was optimized by the relax application (Section \ref{subsubsec:MM_Relax}) that produced 50 structures of a single variant, for each of these structures property scores where available of which all quartiles have been used as a machine learning feature.
of which the quartiles are used as a learning feature. Probe (Section \ref{subsec:MM_Probe}) calculated the solvent accessible surface area (SASA/ACCP) of a protein in square {\AA}ngstrom ({\AA}$^2$) which was used as a structural machine learning feature.  
The sequence features of VIPUR were produced by PSI-blast (Section \ref{subsec:MM_PSI_BLAST}) on non mutated sequences and blasted against the NCBI protein database (nr) which resulted in a position specific scoring matrix (PSSM). 
From the PSSM scores of the information content, non-mutated, mutated, the difference in scores between non-mutated and mutated and the difference between groups \cite{baugh_robust_2016, poultney_rational_2011} used as sequential feature for VIPUR.
% Robust classification of protein variation using structural modelling and large-scale data integration, Baugh et al.
% Rational Design of Temperature-Sensitive Alleles Using Computational Structure Prediction, Poultney et al.
With 106 features generated by the mentioned tools deleteriousness of a protein variant is determined with sparse logistic regression. The term sparse implies that a limited set of features was used because the weights "shrink" to 0 with regularization \cite{baugh_supplementary:_2016}.
%SUPPLEMENTARY: Robust classification of protein variation using structural modelling and large-scale data integration, Baugh et al.
\label{subsec:MM_VIPUR}

\subsection{VIPUR specific tools}

\subsubsection{PSI-BLAST}
Position specific iterative basic local alignment search tool (PSI-BLAST) focuses on distant relatives of proteins by making a profile of the sequence and querying it at a protein sequence database. With the generated results a new profile is constructed which is queried to the same database as the previous query. These steps are repeated several times to determine which residues are found in distant relatives and results is a position specific scoring matrix (PSSM) that describes the frequency of which residues are substituted by a specific other residue \cite{ncbi_psiblast_nodate,ncbi_pssm_nodate,wikipedia_blast_2019}.
%PSIBLAST, NCBI
%PSSM Viewer, NCBI
%BLAST, Wikipedia 
From the PSSMs sequences features were acquired for the VIPUR machine learning method.
\label{subsec:MM_PSI_BLAST}
\newline
\textit{Position-Specific Iterated BLAST 2.7.1+}

\subsubsection{Probe}
Probe is able to evaluate atom packing for a single protein or for interacting proteins. It does this by creating a probe, which is described as a sphere like object, that marks an area with dots when at least two non-covalent atoms are in contact with the probe. \cite{word_visualizing_1999, richardson_lab_probe_nodate}. VIPUR used this tool to calculate solvent accessible surface area (SASA or ACCP).
%Visualizing and quantifying molecular goodness-of-fit: small-probe contact dots with explicit hydrogen atoms11Edited by J. Thornton,Word
%Probe Software : Kinemage Website, Richardson LAB
\label{subsec:MM_Probe}
\newline
\textit{version 2.16.130520}

\subsection{Rosetta}
Rosetta is a software suite that has a variety of tools that are developed to aid in macro molecular and antibody, analysis, design and prediction \cite{rosetta_commons_about_nodate}.
% About | RosettaCommons, Rosetta Commons
However no tools in the suite have been encountered that could introduce missense mutations in the proteins and was done by other software (Sections \ref{subsec:MM_PyMOL}, \ref{subsec:MM_PyRosetta}, \ref{subsec:MM_Modeller}). With the introduced mutations water had to be removed because some tools cannot predict structures well with: water, metals and amino acids that are no part of the standard (20) amino acids \cite{rosetta_commons_how_nodate}.
%How to prepare structures for use in Rosetta, Rosetta commons

Within the tools from Rosetta various scores are assigned to different properties related to bonds, interactions, energies and geometries within structures and are written into a score file. 
From all different scoring metrics the Rosetta score, Rosetta energy unit (REU) or total\_score in the score files, can be used to compare models from the same protein with the same tool. 
Not only is the score based on energy but also it has statistical terms which influence the score based on known favorable folds from existing structures that reside in the curated Rosetta database \cite{shourya_scoring_nodate}. 
In summary to asses models, a lower Rosetta score makes a more natural model.
%Scoring Tutorial, Shourya et al.
\label{subsec:MM_Rosetta}
\newline
\textit{Rosetta software suite Version 3.10}

\subsubsection{Relax}
The Relax application was used by VIPUR and by SPVAA to relax the side chains to minimize energy levels within the local conformational search space \cite{rosetta_commons_relax_nodate} of the structure. 
It determines the energy levels with a Monte Carlo method (Section \ref{section:Chap_Monte_Carlo}) and after a certain set of moves it produces a structure and starts anew \cite{conway_relaxation_2014,tyka_alternate_2011}. 
%Relax application, Rosetta Commons 
%Relaxation of backbone bond geometry improves protein energy landscape modeling, Conway et al.
%Alternate States of Proteins Revealed by Detailed Energy Landscape Mapping, Tyka et al.
\label{subsubsec:MM_Relax}

\subsubsection{DDG Monomer}
DDG monomer is meant to predict energetic stability of a point mutation in  monomeric protein. 
The application was used by VIPUR to collect features related to energies bonds, bridges  and constraints differences between the wild type and a mutated protein. 
To execute the tool a script had to be ran which renumbered the wild type pdb file and it required a "mutation file" that described the changes of a residues based on name and position\cite{leaver-fay_ddg_monomer_nodate}.
%ddg_monomer application, Andrew Leaver-Fay
\label{subsubsec:MM_DDG_Monomer}

\subsubsection{Rescore}
With this tool Rosetta scores can be calculated based on PDB files proteins structures \cite{jared_score_2016} , the output is identical to that is written within the score files produced by Relax (Section \ref{subsubsec:MM_Relax}).
%Score Commands, Jared
\label{subsubsec:MM_Rescore}

\subsubsection{Backrub}
The backrub application is based on the Monte Carlo method (Section \ref{section:Chap_Monte_Carlo}), and alters a protein by moving its backbone residues with a strategy called fix end move (FEM). 
With this strategy, groups of residues are selected at random from the structure that can contain up to: four dihedral, two bond angles and two end points. 
Both ends of a group are fixated at their position in which a new angle $\alpha$ arises, within this angle residues are pivoted in their natural occurring maximum range of $\pm \ang{10}$ \cite{betancourt_efficient_2005, smith_backrub_nodate}.
%Efficient Monte Carlo trial moves for polypeptide simulations, Betancourt
%Backrub application, Smith
With this application backbones of newly introduced mutations were altered, for each attempt a new file was generated and the scores were written to a score file. The lowest Rosetta scoring model was selected to undergo side chain relaxation with the Relax application (Section \ref{subsubsec:MM_Relax}).
\label{subsubsec:MM_Backrub}

\subsection{Structure prediction web services}

\subsubsection{Robetta prediction server}
The web tool Robetta integrates several tools to form protein structures with homology modeling (Section \ref{subsec:GD_Protein_modeling_techniques}).
Its only requirement is an amino acid sequence, but optionally constrains and fragments can be added to disallow movement of certain structures or add known fragments to avoid calculating pieces that are already known. With this information Robetta searches with the help of sequence aligners for known fragments and tries to incorporate them into a single protein structure \cite{song_high-resolution_2013,soding_protein_2005,kallberg_template-based_2012,yang_improving_2011,ovchinnikov_protein_2017}.
%High-Resolution Comparative Modeling with RosettaCM, Song
%Protein homology detection by HMM-HMM comparison, Soding
%Template-based protein structure modeling using the RaptorX web server,KÃ¤llberg
%Improving protein fold recognition and template-based modeling by employing probabilistic-based matching between predicted one-dimensional structural properties of query and corresponding native properties of templates, Yang
%Protein structure determination using metagenome sequence data,Ovchinnikov
The known fragments of TNFRSF1A (Section \ref{subsec:GD_Protein_modeling_techniques}) were given as a template to Robetta and modeled into a whole protein to make it possible to introduce mutations and predict pathogenicity of a variants.\\
\label{subsec:MM_Robetta}
\textit{http://new.robetta.org/}

\subsubsection{I-TASSER prediction server}
The I-TASSER web server is a tool that is able to predict protein structures with a FASTA sequence. The first step it takes is finding structural templates which resemble the sequence by local meta-threading server (LOMETS).
LOMETS starts with multiple sequence alignment of which several sequences will undergo protein threading by different programs to form structural templates. 
The templates are assessed made from: the highest alignment Z-score, a program within LOMETS specific confidence score and sequence identity \cite{zhang_lab_lomets_nodate, wu_lomets:_2007}.
%LOMETS, Zhang LAB
%LOMETS: A local meta-threading-server for protein structure prediction, Wu
The known fragments of TNFRSF1A (Section \ref{subsec:GD_Protein_modeling_techniques}) were given as a template to I-TASSER and modeled into a whole protein to make it possible to introduce mutations and predict pathogenicity of a variants.
\label{subsec:MM_I_TASSER}
\newline
\textit{Server version, https://zhanglab.ccmb.med.umich.edu/I-TASSER/}

\subsubsection{HOPE}
Have yOur Protein Explained (HOPE) is a web service that collects information of about a user specified missense mutation in a protein and comes from various sources. Uniprot (Section \ref{subsec:MM_Uniprot}) is queried with BLAST to find homologous sequences and structures, other features that are found on Uniprot are active sites, domains and various other sequence features that help to identify the function of a region. From the BLAST results homology models are made with Yasara that are sent of to WHAT IF web services that calculate structural information about the protein. Before the formation of a report all information is put into a decision tree to asses mutational effects in context of: contacts, structural locations, non-structural features, previous variant information and amino acid properties. \cite{venselaar_protein_2010,cmbi_hope_nodate,cmbi_hope_nodate-1,cmbi_hope_nodate-2}.
%Protein structure analysis of mutations causing inheritable diseases. An e-Science approach with life scientist friendly interfaces, Venselaar
%HOPE, CMBI
%HOPE about, CMBI
%HOPE methods, CMBI
With this method it is not possible to asses ligands and complexes at once but only a single missense mutation within a protein.
\label{subsec:MM_HOPE}
\textit{Version 1.1.1, https://www3.cmbi.umcn.nl/hope/}

\subsection{Structural modification and visualization software}

\subsubsection{Modeller}
The Modeller software that is developed for homology modeling but it was used for its utilities. Which allowed to complete protein data bank (PDB) structures with missing atoms, predict disulfide bonds that were missing and mutate protein residues \cite{modeller_about_nodate,eswar_comparative_2006,sali_comparative_1993,fiser_modeling_2000}. 
%About MODELLER, Modeller
%Comparative Protein Structure Modeling Using Modeller, Eswar
%Comparative Protein Modelling by Satisfaction of Spatial Restraints, Sali
%Modeling of loops in protein structures., Fiser
\label{subsec:MM_Modeller}
\newline
\textit{Version 9.21}

\subsubsection{PyMOL}
Visualization of 3D structures, making images of proteins, putting monomeres in the correct position, replacing TNF $\beta$ structure with a TNF $\alpha$ in the bound structure and aligning the structures to measure the distance between models and X-ray structures were done with PyMOL \cite{schrodinger_pymol_nodate}. PyMOL was in VIPUR used in combination with Python (Section \ref{subsec:MM_Python}) to perform mutagenesis on the protein structures.
% PyMOL | pymol.org, SchrÃ¶dinger
\label{subsec:MM_PyMOL}
\newline
\textit{Version 2.2.3}

\subsubsection{PyRosetta}
Is an application programming (API) which has Python bindings (Section \ref{subsec:MM_Python}) for the Rosetta software suite (Section \ref{subsec:MM_Rosetta}) and founds its use in VIPUR when no PyMOL (Section \ref{subsec:MM_PyMOL}) was available to mutate residues in a structure \cite{jeffrey_pyrosetta_nodate}.
%PyRosetta, Jeffry
\label{subsec:MM_PyRosetta}
\newline
\textit{Version 4}

\subsection{Amino acid sequence variant tables}

\subsubsection{GAVIN Machine Learning Data Table}
Is a collection of nucleotide mutations from rare diseases used by the GAVIN \cite{van_der_velde_gavin:_2017} machine learning approach. From this set the genes of TNFRSF1A (Section \ref{section:Chap_Cell_Death}) with a missense mutation were filtered (Section \ref{subsec:MM_R}) and written into a format which the variant effect predictor could (VEP) \cite{ensembl_variant_nodate} could read and translate from nucleotide to protein mutations. The classification of the variants was done by experts based on the five tier IARC classification system \cite{plon_sequence_2008}. 
%GAVIN: Gene-Aware Variant INterpretation for medical sequencing, Van der Velde
%Variant Effect Predictor - Homo sapiens - GRCh37 Archive browser 96, ensembl
%Sequence variant classification and reporting: recommendations for improving the interpretation of cancer susceptibility genetic test results, Plon
\label{subsec:MM_GAVIN_data_table}

\subsubsection{gnomAD}
The gnomAD database consists of unified data from large scale genome sequencing data projects and is based on genome reference consortium human genome build 37 human genome 19 (GRCh37/hg19). From this database missense mutations were collected for TNFRSF1A (Section \ref{subsec:CD_TNFRSF1A}), no classification was known from these mutations \cite{gnomad_gnomad_nodate}.
%gnomAD,gnomAD
\label{subsec:MM_GnomAD}

\subsubsection{Infevers}
Infevers is a website about auto hereditary inflammatory diseases for which each are tables that contain information about mutations and their classification. The table for TRAPS (Section \ref{subsec:CD_TNFRSF1A}) was used to collect missense mutations of TNFRSF1A gene \cite{sarrauste_de_menthiere_infevers:_2003}.
%INFEVERS: the Registry for FMF and hereditary inflammatory disorders mutations, Sarrauste de MenthiÃ¨re
\label{subsec:MM_Infevers}

\subsection{Protein functional and structural databases}

\subsubsection{Research Collaboratory for Structural Bioinformatics}
Research Collaboratory for Structural Bioinformatics (RCSB) is a database where whole or fragmented experimentally determined proteins structures, that are published, can be found and downloaded. The Fragments for modeling (Sections \ref{subsec:MM_Robetta}, \ref{subsec:MM_I_TASSER}) whole TNFRSF1A (Section \ref{subsec:CD_TNFRSF1A}) (1EXT \cite{naismith_structures_1996}) and determining the differences in energy levels (Section \ref{subsubsec:MM_Relax}) with TNF $\alpha$ -$\beta$ (1TNR \cite{banner_crystal_1993}) with the interaction site were acquired from this database \cite{burley_rcsb_2018}.
%Structures of the extracellular domain of the type I tumor necrosis factor receptor,Naismith
%Crystal structure of the soluble human 55 kd TNF receptor-human TNFÎ² complex: Implications for TNF receptor activation, Banner
%RCSB Protein Data Bank: Sustaining a living digital data resource that enables breakthroughs in scientific research and biomedical education, Burley
\label{subsec:MM_RCSB}

\subsubsection{Uniprot}
Knowledge from various omic domains about proteins have been linked together into single database called Uniprot which makes all information accessible at once. For TNFRSF1A (Section \ref{section:Chap_Cell_Death}) the FASTA sequences were collected from Uniprot and for structures it redirected to (Section \ref{subsec:MM_RCSB}) \cite{the_uniprot_consortium_uniprot:_2015}.
%UniProt: a hub for protein information,The UniProt Consortium
\label{subsec:MM_Uniprot}

\subsection{Scripting languages}

\subsubsection{Python}
Both VIPUR and the single protein variant analysis approach (SPVAA) were written in Python. SPVAA was written in Python because of its ease of use and the modeller bindings (Section \ref{subsec:MM_Modeller}) that were available. 
The mutations that were put together from the different tables (Sections \ref{subsec:MM_GnomAD}, \ref{subsec:MM_Infevers}, \ref{subsec:MM_GAVIN_data_table}) with R (Section \ref{subsec:MM_R}) were filtered by a Python script. 
From the mutations a compact list was made by a different script that described the chains that had to be altered by modeller to introduce the appropriate mutation into a PDB file. From these PDB files several were selected to be optimized by pipeline that used backrub (Section \ref{subsubsec:MM_Backrub}) and Relax (Section \ref{subsubsec:MM_Relax} ) to optimize the structure.
\label{subsec:MM_Python}
\newline
\textit{Laptop version 2.7.15}
\newline
\textit{Server version 2.7.11}

\subsubsection{R scripting language}
With R the tables from gnomAD, GAVIN and Infevers (Sections \ref{subsec:MM_GnomAD}, \ref{subsec:MM_GAVIN_data_table}, \ref{subsec:MM_Infevers}) of TNFRSF1A missense mutations (Section \ref{subsec:CD_TNFRSF1A}) were merged together in a new comma seperated values file with their known classifications. Ordering and filtering the double mutations and removing double classifcations where done with Python (Section \ref{subsec:MM_Python}). It has also been used in combination ggplot2 \cite{wickham_create_nodate} and data.table \cite{dowle_rs_2019} to make density plots of all scores acquired from Rosetta Backrub and Relax (See the supplementary for R package versions).
%ggplot Wickham
% data table, Dowl et al
\label{subsec:MM_R}
\newline
\noindent
\textit{R scripting front-end version 3.5.2 (2018-12-20)}

\subsection{Utility software}

\subsubsection{Bash}
Unix like operating systems (OS) have a shell which allows users to interact with programs on a computer or with the computer itself based on commands submitted.  The default shell for MacOS and also for several Linux distributions is the Bourne again shell (Bash) which was used to launch Python scripts (Section \ref{subsec:MM_Python}) and submit jobs to the SLURM workload manager (Section \ref{subsec:MM_SLURM}). 
\label{subsec:MM_Bash}
\newline
\textit{Laptop Version GNU bash, version 3.2.57(1)-release (x86\_64-apple-darwin18)}
\newline
\textit{Server Version GNU bash, version 4.1.2(2)-release (x86\_64-redhat-linux-gnu)}

\subsubsection{SLURM}
For computational jobs where a laptop or desktop does not suffice because due to the lack computational resources a computer cluster could come to aid.
These clusters consist out of several computers that execute resource intensive tasks, to manage these systems as optimal and fair as possible a workload manager like simple Linux utility resource management (SLURM), is installed. Jobs are submitted that request resources for execution and are scheduled on the systems queue.
\label{subsec:MM_SLURM}

\subsubsection{MPI}
Some tools from the Rosetta software suite (Sections \ref{subsec:MM_Rosetta}) have the ability to use multiple central processing unit (CPU) cores from a single computer or from multiple computers. With a message parsing interface (MPI) it is possible for software to communicate between CPU cores on the same and on different computers to exchange information about processes giving the ability to share work between computers and CPUs.
\label{subsec:MM_MPI}
\newline
\textit{OpenMPI/1.8.8-GNU-4.9.3-2.25}
\newline



