
\subsection{Two methods:scale and detail}
Protein variants were analyzed in large groups by the VIPUR approach by several tools to which machine learning was applied. Within the approach proteins that had no experimental structure available were homology modeled with Modeller (Section \ref{subsec:MM_Modeller}), parts and whole structures from proteins were collected from Modbase \cite{} and
% modbase, a database of annotated comparative protein structure models and associated resources, pieper
 SWISS-MODEL \cite{} .
% Automated comparative protein structure modeling with SWISS-MODEL and Swiss-PdbViewer: A historical perspective, Guex
% Modeling protein quaternary structure of homo- and hetero-oligomers beyond binary interactions by homology, Bertoni
% Toward the estimation of the absolute quality of individual protein structure models, Benkert
% The SWISS-MODEL Repository—new features and functionality, Bienert
% SWISS-MODEL: homology modelling of protein structures and complexes, Waterhouse


 The structures in this approach were acquired rom the structures that were acquired 

\subsection{Rosetta}
For the prediction and analysis of protein structures the Rosetta software suite was used, it contains various tools for protein, antibody analysis and design \cite{}.
% About | RosettaCommons, Rosetta Commons
The scores generated for the machine learning within the VIPUR approach rely on results generated by Rosetta software and to apply this approach the steps are reproduced.  
Several strategies were employed for realizing mutated structures, the first strategy was to identify the whole structure of proteins



\subsubsection{Relax}
\label{subsubsec:MM_Relax}

\subsubsection{Abinitio}
\label{subsubsec:MM_Abinitio}

\subsubsection{Backrub}
\label{subsubsec:MM_Backrub}

\subsubsection{Rescore}
\label{subsubsec:MM_Rescore}

\label{subsec:MM_Rosetta}

\subsection{BLAST}
\label{subsec:MM_BLAST}

\subsection{PSI-BLAST}
\label{subsec:MM_PSI_BLAST}


\subsection{Probe}
\label{subsec:MM_Probe}

\subsection{Modeller}
\label{subsec:MM_Modeller}

\subsection{RCSB}
\label{subsec:MM_RCSB}

\subsection{Uniprot}
\label{subsec:MM_Uniprot}




 The initial structure of the protein was produced with the application abinitio relax. For the prediction the application requires an amino acid sequence to identify homologous sequences in a curated database. Homologous sequences within the database are found by the BLAST algorithm, when a

For the search of the sequences it uses the BLAST algorithm and to find homologous amino acid sequences which have protein structures.

requires an amino acid sequence and  it takes an amino acid sequence as input and searches in a curated protein database BLAST for finding homologous sequences. 


to align sequences with to acquire homologous sequences.  The homologous With these sequences it finds structures related to the protein
For the prediction of the initial structure of TNFR the application abinitio relax was used. 

With this tool a sequence is inserted as input that is aligned to 
 


\subsection{I-TASSER}
\label{subsec:MM_I_TASSER}

\subsection{PyMOL}
Visualization of 3D structures, making images of proteins and putting the known orientations of monomeres in position  were done in PyMOL \cite{}.
% PyMOL | pymol.org, Schrödinger
Since some protein structures consist of multiple identical monomers they are left out of the structure and supplied with information about how the monomers are position to form the whole oligomer structure (Sections \ref{subsec:MM_RCSB}, \ref{subsec:MM_Uniprot}).

