\subsection{Two methods: scale and detail}

VIPUR is a relative large scale approach for analyzing protein variants and predicts whether a protein variant is pathogenic or benign, for its machine learning features it relies on

 using machine learning, for its features it relies on 

VIPUR is a relative large scale approach for analyzing protein variants using machine learning, for the features used for ma relies on DDG monomer (Section \ref{subsubsec:MM_DDG_Monomer}), Relax (Section \ref{subsubsec:MM_Relax}) and Rescore (Section \ref{subsubsec:MM_Rescore}) from the Rosetta software suite (Section \ref{subsec:MM_Rosetta}). The remaining features are collected from PSI-BLAST (Section \ref{subsec:MM_PSI_BLAST}) and Probe (Section \ref{subsec:MM_Probe}). 
Features from the proteins acquired by analyzing the structures with Probe and Rosetta were collected from Modbase \cite{} and
% modbase, a database of annotated comparative protein structure models and associated resources, pieper
SWISS-MODEL \cite{}.
% Automated comparative protein structure modeling with SWISS-MODEL and Swiss-PdbViewer: A historical perspective, Guex
% Modeling protein quaternary structure of homo- and hetero-oligomers beyond binary interactions by homology, Bertoni
% Toward the estimation of the absolute quality of individual protein structure models, Benkert
% The SWISS-MODEL Repository—new features and functionality, Bienert
% SWISS-MODEL: homology modelling of protein structures and complexes, Waterhouse
Not all structures that were used from the databases for machine learning were complete, there were structures of which only pieces were experimentally determined and for some no structure was available. To level this problem homology models were made with Modeller (Section \ref{subsec:MM_Modeller} based on known homologous structures. In some experimental determined structures duplicate chains, ligands, metals and non-standard amino acids were present, these inconsistencies were able to alter the features generated by software and could in some case hinder feature collection, were therefore removed. VIPUR introduced mutations within the protein chains with PyMOL (Section \ref{subsec:MM_PyMOL}) by default or PyRosetta (Section \ref{subsec:MM_PyRosetta}) if PyMOL was not available.

Another approach for determining pathogenicity of a mutation is by assessing energy differences between a wild type and mutant protein residues inside its complex. Analyzing mutations from this perspective gives the ability to view a complex in whole and determine how residues cause perturbations in a complex. Missense mutations in monomers of complexes were made with Modeller (Section \ref{subsec:MM_Modeller}) and the backbone was refined with Rosetta's backrub application (Section \ref{subsubsec:MM_Backrub}), to lower the energy levels within side chains Rosetta relax (Section \ref{subsubsec:MM_Relax}). This method was tested with TNFRSF1A (Section \ref{subsec:CD_TNFRSF1A}) and its ligands TNF $\alpha$ and $\beta$. Al tough this method shows similarities to VIPUR's pre-processing it keeps most elements which VIPUR excludes. 

%This approach shows similarities to pre-processing of the VTS, but it still has ligands, duplicate residues and metals, water was left out since it could cause errors to occur during the modeling.
%An aspect that should be identified is how the mutated protein will interact with complex and ligands that interact with the protein.  with this approach the interaction of the product within the complex is taken into account since changes within a product should be determined in whole since they can be so minimal and none harmful at first sight but could be devastating when part of the complex\cite{sorcecryXXXXXXXXXXXXXX}.
%and determining the effects of a mutation by inserting, replacing or removing a residue and remodeling the structure. With this step extra information has to be added how the protein forms a complex or interacts with its ligands to determine the effect of the mutation.
% which has been done on the protein TNFRSF1A (Section \ref{subsec:CD_TNFRSF1A}). 
%Interaction with protein and ligand (TNFRSF1A TNF)
%The experimentally determined structure from TNFRSF1A (Section \ref{subsec:CD_TNFRSF1A}) has been acquired from Uniprot (Section \ref{subsec:MM_Uniprot}) with it interacting with TNF $\beta$ (Section \ref{subsec:CD_TNF_A_B}) from pr
With the VIPUR approach treating each protein similar by its current methods an alternative method  the same by applying each method on all structures and   a different method an uncertain how it will behave a more traditional approach was also take.
A different approach that was taken by investigating a single protein (Section \ref{}) structure and itsthorough investigation of changes within the

Models within the VIPUR training set (VTS) had different
Models acquired from t
Proteins within the VIPUR training set (VTS) were a full structure had only parts or none
Not every protein used within the VIPUR Training Set (VTS) had a whole structure or 

Within the approach proteins that had no experimental structure available were homology modeled with Modeller (Section \ref{subsec:MM_Modeller}), parts and whole structures from proteins were collected from 
Another approac
Another approach that was taken was to thoroughly investigate 

Protein variants were analyzed in large groups by the VIPUR approach by several tools to which machine learning was applied. 

\subsection{Rosetta}
For the prediction and analysis of protein structures the Rosetta software suite was used, it contains various tools for protein, antibody analysis and design \cite{}.
% About | RosettaCommons, Rosetta Commons
The scores generated for the machine learning within the VIPUR approach rely on results generated by Rosetta software and to apply this approach the steps are reproduced.  
Several strategies were employed for realizing mutated structures, the first strategy was to identify the whole structure of proteins



\subsubsection{Relax}
\label{subsubsec:MM_Relax}

\subsubsection{DDG Monomer}
\label{subsubsec:MM_DDG_Monomer}

\subsubsection{Rescore}
\label{subsubsec:MM_Rescore}

\subsubsection{Abinitio}
\label{subsubsec:MM_Abinitio}

\subsubsection{Backrub}
\label{subsubsec:MM_Backrub}

\label{subsec:MM_Rosetta}

\subsection{PyRosetta}
\label{subsec:MM_PyRosetta}

\subsection{BLAST}
\label{subsec:MM_BLAST}

\subsection{PSI-BLAST}
\label{subsec:MM_PSI_BLAST}

\subsection{Probe}
\label{subsec:MM_Probe}

\subsection{Modeller}
\label{subsec:MM_Modeller}

\subsection{RCSB}
\label{subsec:MM_RCSB}

\subsection{Uniprot}
\label{subsec:MM_Uniprot}




 The initial structure of the protein was produced with the application abinitio relax. For the prediction the application requires an amino acid sequence to identify homologous sequences in a curated database. Homologous sequences within the database are found by the BLAST algorithm, when a

For the search of the sequences it uses the BLAST algorithm and to find homologous amino acid sequences which have protein structures.

requires an amino acid sequence and  it takes an amino acid sequence as input and searches in a curated protein database BLAST for finding homologous sequences. 


to align sequences with to acquire homologous sequences.  The homologous With these sequences it finds structures related to the protein
For the prediction of the initial structure of TNFR the application abinitio relax was used. 

With this tool a sequence is inserted as input that is aligned to 
 


\subsection{I-TASSER}
\label{subsec:MM_I_TASSER}

\subsection{PyMOL}
Visualization of 3D structures, making images of proteins and putting the known orientations of monomeres in position  were done in PyMOL \cite{}.
% PyMOL | pymol.org, Schrödinger
Since some protein structures consist of multiple identical monomers they are left out of the structure and supplied with information about how the monomers are position to form the whole oligomer structure (Sections \ref{subsec:MM_RCSB}, \ref{subsec:MM_Uniprot}).
\label{subsec:MM_PyMOL}

