\subsection{VIPUR approach}
Variant interpretation using Rosetta (VIPUR) is a machine learning approach for predicting deleteriousness of proteins (loss of function) and uses sequential and structural information. To train VIPUR a training set was made that contains sequence and structure features. Structures within the VIPUR training set (VTS) were collected  from Modbase \cite{} and
% modbase, a database of annotated comparative protein structure models and associated resources, pieper
SWISS-MODEL \cite{}.
% Automated comparative protein structure modeling with SWISS-MODEL and Swiss-PdbViewer: A historical perspective, Guex
% Modeling protein quaternary structure of homo- and hetero-oligomers beyond binary interactions by homology, Bertoni
% Toward the estimation of the absolute quality of individual protein structure models, Benkert
% The SWISS-MODEL Repositoryâ€”new features and functionality, Bienert
% SWISS-MODEL: homology modelling of protein structures and complexes, Waterhouse
Proteins that did not have a structure within the VTS were modeled with Modeller (Section \ref{subsec:MM_Modeller} based on protein fragments that had the highest amino acid sequence identity to the protein. Some structures from the databases had: duplicate chains, ligands, metals and non-standard amino acids which were removed to avoid inconsistencies that could alter the features generated by the tools and hinder feature collection within Rosetta tools (Section \ref{subsec:MM_Rosetta}), therefore they were removed to make the data homogeneous.  
For all proteins a mutation file was made that described where and which residues had to be mutated, with this file DDG monomer (Section \ref{subsubsec:MM_DDG_Monomer}) can predict changes in the protein structure of which all features are used for prediction. Structural mutations of proteins that are in the VTS were introduced by a script using PyMOL (Section \ref{subsec:MM_PyMOL}) by default or PyRosetta (Section \ref{subsec:MM_PyRosetta}) if PyMOL was not available. After mutation the structure is optimized by the relax application (Section \ref{subsubsec:MM_Relax}) and produces 50 relaxed structures of a single variant where the properties of each protein are written to a score file of which the quartiles are used as a learning feature. Probe (Section \ref{subsec:MM_Probe})calculated the solvent accessible surface area (SASA/ACCP) of a protein in square {\AA}ngstrom ({\AA}$^2$) which is a structural machine learning feature.  The sequence features of VIPUR are produced by PSI-blast (Section \ref{subsec:MM_PSI_BLAST}) on non mutated sequences and blasted against the NCBI protein database (nr) which results in a position specific scoring matrix (PSSM). From the PSSM scores of the non-mutated, mutated, the difference in scores between non-mutated and mutated, information content and the difference between groups \cite{} are all sequential features for VIPUR.
% Robust classification of protein variation using structural modelling and large-scale data integration, Baugh et al.
% Rational Design of Temperature-Sensitive Alleles Using Computational Structure Prediction, Poultney et al.
With 106 features generated by the mentioned tools deleteriousness of a protein variant is determined with sparse logistic regression. The term sparse implies that a limited set of features was used because the weights "shrink" to 0 with regularization \cite{}.
%SUPPLEMENTARY: Robust classification of protein variation using structural modelling and large-scale data integration, Baugh et al.
\label{subsec:MM_VIPUR}

\subsection{Simple protein variant analysis approach}
Another method for determining function loss in a protein variant is through assessment of difference in energy levels between a wild type and a variant in the complex where it resides. Analyzing mutations from this perspective gives the ability to view a complex in whole and determine how residues cause perturbations in a complex. To make a variant of the wild type, a structure was required wherein a missense mutation could be introduced with Modeller (Section \ref{subsec:MM_Modeller}). The backbone structure of the variant was modified with the backrub application (Section \ref{subsubsec:MM_Backrub}) to make it better interact with other amino acid backbones in the structure resulting in 1000 models. The lowest Rosetta scoring (Section \ref{subsec:MM_Rosetta}) one would be selected to further improve the side chains with the Relax application
 (Section \ref{subsubsec:MM_Relax}) which makes the side chains of the protein move into lower energy formations. From these 64 models were made where its energies levels were compared to the native structure to determine the effects a mutation would have on the protein, the lowest scoring ones were used as figures. This method shows similarities to that of VIPUR and was only tested on TNFRSF1A (Section \ref{subsec:CD_TNFRSF1A}) and its ligands TNF $\alpha$ and $\beta$. This method keeps: duplicate chains and protein ligands within the structure, water is excluded since it can cause issues with Rosetta tools (Section \ref{subsec:MM_Rosetta}). This procedure with these steps is defined as the Simple protein variant analysis approach (SPVAA).
\label{subsec:MM_SPVAA}

\subsection{Rosetta}
Rosetta is a software suite that has a variety of tools that are developed to aid in macro molecular and antibody ,analysis, design and prediction \cite{}.
% About | RosettaCommons, Rosetta Commons
However no tools in the suite have been encountered that could introduce missense mutations in the proteins and has been dealt with by other software (Sections \ref{subsec:MM_PyMOL}, \ref{subsec:MM_PyRosetta}, \ref{subsec:MM_Modeller}). With the introduced mutations water had to be removed because some tools cannot predict structures well with: water, metals and amino acids that are no part of the standard (20) amino acids \cite{}.
%How to prepare structures for use in Rosetta, Rosetta commons

Within the tools from Rosetta various scores are assigned to different properties related to bonds, interactions, energies and geometries within structures and are written to a score file. From all different scoring metrics the Rosetta score or Rosetta energy unit (REU) , which is the total\_score in the score files, can be used to compare models made from the same protein per tool. Not only is the score based on energy but also it has statistical terms which influence the score based on known favorable folds from existing structures that reside in the curated Rosetta database \cite{}. In summary a lower Rosetta score would make a more natural model.
%Scoring Tutorial, Shourya et al.
\label{subsec:MM_Rosetta}
\newline
\textit{Rosetta software suite Version 3.10}

\subsubsection{Relax}
The Relax application was used by VIPUR and by SPVAA to relax the side chains to minimize energy levels within the local conformational search space \cite{} of the structure, it determines the most likely energy levels with the Monte Carlo method (Section \ref{section:Chap_Monte_Carlo}) and after a certain set of moves it produces a structure and stars anew\cite{}. Scores from each produced by Relax were written into a single score file.
%Relax application, Rosetta Commons 
%Relaxation of backbone bond geometry improves protein energy landscape modeling, Conway et al.
%Alternate States of Proteins Revealed by Detailed Energy Landscape Mapping, Tyka et al.
\label{subsubsec:MM_Relax}

\subsubsection{DDG Monomer}
DDG monomer is meant to predict energetic stability of a point mutation in  monomeric protein.The application was used by VIPUR to collect features related to energies and hydrogen, disfulfide, bonds and constraints differences between the wild type and a mutated protein. To execute the tool a script had to be ran that renumbers the wild type pdb file and it requires a "mutation file" that describes the change of a residue based on name and position changes to a different residue \cite{}.
%ddg_monomer application, Andrew Leaver-Fay
%Backrub application, Smith
\label{subsubsec:MM_DDG_Monomer}

\subsubsection{Rescore}
With this tool Rosetta scores can be calculated based on silent or PDB files proteins structures \cite{} , the output is identical to that is written within the score files produced by Relax (Section \ref{subsubsec:MM_Relax}).
%Score Commands, Jared
\label{subsubsec:MM_Rescore}

\subsubsection{Backrub}
The backrub application is based on the Monte Carlo method (Section \ref{section:Chap_Monte_Carlo}), and alters a protein by moving its backbone residues with a strategy called fix end move (FEM). With this strategy, groups of residues are selected at random from the structure, it can contain up to: four dihedral, two bond angles and two end points. Both ends of a group are fixated at their position in which a new angle $\alpha$ arises, within this angle residues are pivoted in their natural occurring maximum range of $\pm \ang{10}$ \cite{}.
%Efficient Monte Carlo trial moves for polypeptide simulations, Betancourt
With this method the backbones of newly introduced mutations were altered, for each attempt a new file was generated and a score was written to a score file, from which the lowest Rosetta scoring was selected to be further relaxed (Section \ref{subsubsec:MM_Relax}).
It was used on the mutated protein to relax the modified backbone structure.
\label{subsubsec:MM_Backrub}

\subsection{PyRosetta}
Is an application programming (API) which has Python bindings (Section \ref{subsec:MM_Python}) for the Rosetta software suite (Section \ref{subsec:MM_Rosetta}), it founds its use in VIPUR when no PyMOL (Section \ref{subsec:MM_PyMOL}) was available to mutate residues within a structure \cite{}.
%PyRosetta, Jeffry
\label{subsec:MM_PyRosetta}
\newline
\textit{Version 4}

\subsection{PSI-BLAST}
Position specific iterative basic local alignment search tool (PSI-BLAST) focuses on distant relatives of proteins by making a profile of the sequence and querying it at a protein sequence database. With the generated results a new profile is constructed and queried again, these steps are repeated several times to determine which residues are found in relatives of the protein. The result is a position specific scoring matrix (PSSM) which describes the frequency of which residues are substituted by a specific other residue, positive is more, negative is less common \cite{}.
%PSIBLAST, NCBI
%PSSM Viewer, NCBI
%BLAST, Wikipedia 
From the PSSMs sequences features were acquired for the VIPUR machine learning method.
\label{subsec:MM_PSI_BLAST}
\newline
\textit{Position-Specific Iterated BLAST 2.7.1+}

\subsection{Probe}
Probe is able to evaluate atom packing for a single protein or interacting proteins by creating a probe, which is described as a sphere like object, that marks an area with dots when at least two non-covalent atoms are in contact with the probe at the same position \cite{}. VIPUR used this tool to calculate solvent accessible surface area (SASA or ACCP).
%Visualizing and quantifying molecular goodness-of-fit: small-probe contact dots with explicit hydrogen atoms11Edited by J. Thornton,Word
%Probe Software : Kinemage Website, Richardson LAB
\label{subsec:MM_Probe}
\newline
\textit{version 2.16.130520}

\subsection{Robetta prediction server}
The web tool Robetta integrates several tools to form protein structures based on sequence alignments of previously discovered structures also known as homology modeling (Section \ref{subsec:GD_Protein_modeling_techniques}).
It requires an amino acid sequence, optionally constrains and fragments can be added to disallow movement of certain structures or add known fragments to avoid calculating pieces that are already known. With this information Robetta search with the help of sequence aligners for known fragments and tries to incorporate them into a single protein structure \cite{}.
%High-Resolution Comparative Modeling with RosettaCM, Song
%Protein homology detection by HMM-HMM comparison, Soding
%Template-based protein structure modeling using the RaptorX web server,KÃ¤llberg
%Improving protein fold recognition and template-based modeling by employing probabilistic-based matching between predicted one-dimensional structural properties of query and corresponding native properties of templates, Yang
%Protein structure determination using metagenome sequence data,Ovchinnikov
\label{subsec:MM_Robetta}

\subsection{I-TASSER prediction server}
The I-TASSER web server is a tool that is able to predict protein structures with a FASTA sequence. The first step it takes is finding structural templates which resemble the sequence by local meta-threading server (LOMETS). LOMETS starts with multiple sequence alignment of which several sequences will undergo protein threading by different programs to form structural templates. The templates are assessed based on the highest alignment Z-score, the program specific confidence score and sequence identity \cite{}.
%LOMETS, Zhang LAB
%LOMETS: A local meta-threading-server for protein structure prediction, Wu
The known fragments of TNFRSF1A (Section \ref{subsec:GD_Protein_modeling_techniques}) were given as a template to I-TASSER and modeled into a whole protein to make it possible to introduce mutations and predict pathogenicity of a variants.
\label{subsec:MM_I_TASSER}
\newline
\textit{Server version}

\subsection{Modeller}
Modeller is software that is developed for homology modeling but it was used for its utilities which allowed to; complete protein data bank (PDB) structures with missing atoms; predict disulfide bonds that were missing and mutate protein residues \cite{}. 
%About MODELLER, Modeller
%Comparative Protein Structure Modeling Using Modeller, Eswar
\label{subsec:MM_Modeller}
\newline
\textit{Version 9.21}

\subsection{GAVIN Machine Learning Data Table}
Is a collection of nucleotide mutations from rare diseases used by the GAVIN \cite{} machine learning approach. From this set the genes of TNFRSF1A (Section \ref{section:Chap_Cell_Death}) with a missense mutation were filtered (Section \ref{subsec:MM_R}) and written into a format which the variant effect predictor could (VEP) \cite{} could read and translate from nucleotide to protein mutations. The classification of these variants was according to Clinvar significance values \cite{}. 
%GAVIN: Gene-Aware Variant INterpretation for medical sequencing, Van der Velde
%Variant Effect Predictor - Homo sapiens - GRCh37 Archive browser 96, ensembl
%Representation of clinical significance in ClinVar and other variation resources at NCBI, NCBI
\label{subsec:MM_GAVIN_data_table}

\subsection{GenomAD}
The GenomAD database consists of unified data from large scale genome sequencing data projects and is based on genome reference consortium human genome build 37 human genome 19 (GRCh37/hg19). From this database missense mutations were collected for TNFRSF1A (Section \ref{subsec:CD_TNFRSF1A}), no classification was known for these mutations \cite{}.
%gnomAD,gnomAD
\label{subsec:MM_GnomAD}

\subsection{Infevers}
Is a website about hereditary auto immune diseases with for each disease a downloadable table about the known mutations and their classification, when classified with pathogenic or benign its function has been observed. The table for TRAPS disease (Section \ref{subsec:CD_TNFRSF1A}) was used to collect missense mutations of TNFRSF1A gene \cite{}.
%INFEVERS: the Registry for FMF and hereditary inflammatory disorders mutations, Sarrauste de MenthiÃ¨re
\label{subsec:MM_Infevers}

\subsection{Research Collaboratory for Structural Bioinformatics}
Research Collaboratory for Structural Bioinformatics (RCSB) is a database where whole or fragmented experimentally determined proteins structures that are published can be found and downloaded. The Fragments for modeling (Sections \ref{subsec:MM_Robetta}, \ref{subsec:MM_I_TASSER}) whole TNFRSF1A (Section \ref{subsec:CD_TNFRSF1A}) (1EXT \cite{}) and determining the differences in energy levels (Section \ref{subsubsec:MM_Relax}) with TNF $\beta$ (1TNR \cite{}) with the interaction site were acquired from this database \cite{}.
%Crystal structure of the soluble human 55 kd TNF receptor-human TNFÎ² complex: Implications for TNF receptor activation, Banner
%Structures of the extracellular domain of the type I tumor necrosis factor receptor,Naismith
%RCSB Protein Data Bank: Sustaining a living digital data resource that enables breakthroughs in scientific research and biomedical education, Burley
\label{subsec:MM_RCSB}

\subsection{Uniprot}
Knowledge from various omic domains about proteins has been linked together into single database called Uniprot which makes all information accessible at once, for TNFRSF1A (Section \ref{section:Chap_Cell_Death}) the FASTA sequences were collected from Uniprot and for structures it redirected to (Section \ref{subsec:MM_RCSB}) \cite{}.
%UniProt: a hub for protein information,The UniProt Consortium
\label{subsec:MM_Uniprot}

\subsection{PyMOL}
Visualization of 3D structures; making images of proteins; putting the known orientations of monomeres in position; replacing TNF $\beta$ structure with a TNF $\alpha$ in the bound structure and aligning the structures to measure the distance between X-ray crystal structures and the produced models were done with PyMOL \cite{}. PyMOL had a different use in VIPUR where it was used in combination with Python (Section \ref{subsec:MM_Python}) to perform mutagenesis on the protein structures to introduce a missense mutations.
% PyMOL | pymol.org, SchrÃ¶dinger
\label{subsec:MM_PyMOL}
\newline
\textit{Version 2.2.3}

\subsection{HOPE}
Have yOur Protein Explained (HOPE) is a web service that collects information of about a user specified missense mutation in a protein and comes from various sources. Uniprot (Section \ref{subsec:MM_Uniprot}) is queried with BLAST to find homologous sequences and structures, other features that are found on Uniprot are active sites, domains and various other sequence features that help to identify the function of a region. From the BLAST results homology models are made with Yasara that are sent of to WHAT IF web services that calculate structural information about the protein. Before the formation of a report all information is put into a decision tree to asses mutational effects in contexts of: contacts, structural locations, non-structural features, previous variant information and amino acid properties to form an automated report \cite{}.
%Protein structure analysis of mutations causing inheritable diseases. An e-Science approach with life scientist friendly interfaces, Venselaar
%HOPE, CMBI
%HOPE about, CMBI
With this method it is not possible to asses ligands and complexes at once but only a single missense mutation within a monomer.
\label{subsec:MM_HOPE}
\textit{Version 1.1.1}

\subsection{Bash}
Unix like operating systems (OS) have a shell which allows users to interact with programs on a computer or with the computer itself based on commands submitted.  The default shell for MacOS and also for several Linux distributions is the Bourne again shell (Bash) which was used to launch Python scripts (Section \ref{subsec:MM_Python}) and submit jobs to the SLURM workload manager (Section \ref{subsec:MM_SLURM}). 
\label{subsec:MM_Bash}
\newline
\textit{Laptop Version GNU bash, version 3.2.57(1)-release (x86\_64-apple-darwin18)}
\newline
\textit{Server Version GNU bash, version 4.1.2(2)-release (x86\_64-redhat-linux-gnu)}

\subsection{Python}
Both VIPUR and the pipeline that minimizes backbone (Section \ref{subsubsec:MM_Backrub}) and side chain energies (Section \ref{subsubsec:MM_Relax}) were written in Python due to its capabilities, ease of use and because modeller (Section \ref{subsec:MM_Modeller}) for MacOS relies on the system version of Python and does currently not support newer versions besides the one found within the OS of Mac. The mutations that were put together from the different tables (Sections \ref{subsec:MM_GnomAD}, \ref{subsec:MM_Infevers}, \ref{subsec:MM_GAVIN_data_table}) with R {Section \ref{subsec:MM_R}) were filtered by a Python script. To apply each mutation correctly on the proteins in the detailed method a script was written in which files were generated that described in a compact format on which chains and position a mutation resided.
\label{subsec:MM_Python}
\newline
\textit{Laptop version 2.7.15}
\newline
\textit{Server version 2.7.11}

\subsection{R scripting language}
With R the tables from GenomeAD, GAVIN and Infevers (Sections \ref{subsec:MM_GnomAD} \ref{subsec:MM_GAVIN_data_table} \ref{subsec:MM_Infevers}) of TNFRSF1A missense mutations (Section \ref{subsec:CD_TNFRSF1A}) were merged together in a new comma seperated values file with their known classifications. Ordering and filtering the double mutations and removing double classifcations where done with Python (Section \ref{subsec:MM_Python}). It has also been used in combination ggplot2 and data.table to make density plots of all scores acquired from Rosetta Backrub and Relax.
\label{subsec:MM_R}
\newline
\textit{R scripting front-end version 3.5.2 (2018-12-20)}

\subsection{SLURM}
For computational jobs where a laptop or desktop does not suffice because due to the lack computational resources a computer cluster could come to aid. These clusters consist out of several computers that execute resource intensive tasks, to manage these systems for many clients and to use these clusters optimal a workload manager mlike simple Linux utility resource management (SLURM), is installed. Jobs are submitted that request resources for execution and are scheduled on the systems queue which is ordered based on priorities, resource requirement and time.
\label{subsec:MM_SLURM}

\subsection{MPI}
Some tools from the Rosetta software suite (Sections \ref{subsec:MM_Rosetta}) have the ability to use multiple central processing unit (CPU) cores from a single computer or from multiple computers. With a message parsing interface (MPI) it is possible for software to communicate between CPU cores on the same and on different computers to exchange information about processes and therefor solving solutions faster.
\label{subsec:MM_MPI}
\newline
\textit{OpenMPI/1.8.8-GNU-4.9.3-2.25}
\newline
