\subsection{Protein modeling techniques}
An alternative approach for determining structures, compared to experimental methods, is protein modeling wherein structures are generated computationally. One of the benefits from making a structure in such a way is that laws of physics do not hinder the collection of structural information, but the lack of physics is also its weakness since the structures tend to be a less accurate representations of proteins.

With the method homology modeling the amino acid sequences of proteins are aligned to sequences of experimental determined structures, based on these alignments a template is formed whereon structural fragments are built. It is not recommended to use this strategy if the sequence identity is less than 20\% since there might not be any structural relation \cite{chothia_relation_1986}.
% The relation between the divergence of sequence and structure in proteins, Chotia and Lesk
Another approach is protein threading which relies on the observation of folds in previously determined experimental structures, with the occurrence of specific folds a probability is predicted that determines how a protein folds in a specific manner. 

Strategies are continuously being developed and improved to determine unknown structures, but all have partially similar guidelines wherein the avoidance of steric hindrance \cite{wikipedia_ramachandran_2019} and low energies, determined by scoring systems \cite{shourya_scoring_nodate}, are important. From the computer generated models many are less accurate than those experimentally determined structures, yet computational models could potentially gain the upper hand in solving membrane proteins\cite{yonath_x-ray_2011}.
% Ramachandran plot, Wikipedia
%Scoring Tutorial , Shourya et al.
%X-ray crystallography at the heart of life science, Yonath
\label{subsec:GD_Protein_modeling_techniques}

\subsection{A theoretical large scale implementation of structural protein variant assessment}
With the wide spectrum of potential different proteins it can be difficult to produce any form of universal protein assessment tool, that is able to determine if a mutation is harmful or not based on structural information. However a first step to solve such a complex problem would be by determining the correct approach. In this case it is assumed that a machine learning would be the best method for detecting patterns in structures and classifying the effect of structural changes, because it has the ability to learn from structural mutations currently available and develop new insights \cite{evans_alphafold:_nodate}.

Since the problem is so complex it should be divided into smaller more feasible problems, beginning by separating the different protein classes, which  for example can be done according to The Structural Classification of Proteins database (SCOP) \cite{andreeva_scop2_2014}. A first discrimination between the proteins could be made based on protein type/fold class (membrane, globular, fibrous and disordered -proteins) because these differences already predetermine some functions and locations for certain proteins in a cell \cite{wikipedia_membrane_2019, wikipedia_globular_2019,wikipedia_scleroprotein_2018,wikipedia_intrinsically_2019}. After formation of these classes each should have its own machine learning method applied so their features can be analyzed within context of where and how they function. The next set of discriminators is highly dependent on the variations in classes, but all have features in the end describing bonds, interactions and movement of complexes in protein structures. When for each of the main classes methods have been developed a meta classifier determines which method should be applied to determine the effect of a mutation.
%SCOP2 prototype: a new approach to protein structure mining, Andreeva et. al.
%Membrane protein, Wikipedia
%Globular protein, Wikipedia
%Scleroprotein, Wikipedia
%Intrinsically disordered proteins, Wikipedia
\label{subsec:GD_theoratical_large_scale_implementation}