\subsection{Mutations and its effects in the central dogma of molecular biology}
Within the human genome mutations occur continuously by internal and external factors that substitute, remove, insert or alter the reading frame in a nucleotide sequence. Mutations are not without consequences and can be: beneficial, benign or in most cases pathogenic because they replace a nucleotide which serves a purpose at the specific position in a sequence. Alterations in sequences might lead to a difference in ribonucleic acid (RNA) transcription rates or differences in the RNA transcript that is formed from the deoxyribonucleic acid (DNA) which both can influence the cellular machinery. Mutations outside a gene could lead to lowered or heightened transcription of a protein, when a mutation resides inside a gene it could lead to proteins that are: unstable during or after formation, perform less optimal or are not functional \cite{}.
%TRAPS, NIH
%Sickle cell disease, NIH
%Cystic fibroses, NIH

\subsection{A general concept of structural levels within proteins and the effect of mutations}
The formation of protein structures is classified in different levels, distinctions are made based on bindings and structures that arise from them. 
The order in which amino acids appear in a sequence is called the primary structure, in this level amino acids are only bound to each other by peptide bonds. 
Within a primary structure amino acids can form new peptide bonds between the N and C -terminus of an amino acid, with these bonds 3D structures are made called $\alpha$ helices and $\beta$-sheets that together make up the secondary structure.
More alterations to a single amino acid sequence in the 3D can come from disulfide bridges, ion, hydrogen -bonds, hydrophobic and hydrophilic -interactions formed by the residues of the amino acids, together these bonds form the tertiary structure.
By combining multiple tertiary structures the quaternary structure of a protein can be formed out of the mentioned bonds, bridges and interactions \cite{}.
% Protein structure, Wikipedia
%Protein Conformation and Diagnostic Tests: The Prion Protein, Bennion

Mutations within proteins can have different effects to protein structures, often single missense mutations often have minimal effect on the backbone of a protein \cite{} but can result in destabilization of the structure when assembled or can disrupt the active site. Frameshift mutations often cause larger disruptions within the structure and often lead to proteins that are deformed or have early stop codons \cite{}.
%Modeling mutations in protein structures, Feyfant et al.
%The relation between the divergence of sequence and structure in proteins., Chothia and Lesk 
%A frameshift mutation in NOD2 associated with susceptibility to Crohn's disease, Ogura et. al.
\label{subsec:GD_Structural_Levels_and_Mutation_Effects}


\subsection{Addition of structural data to diagnosis and treatment in healthcare}
Acquiring information about DNA sequences highly relies on experimental sequencing methods and became cheaper over the years \cite{} and found its use in diagnosing patients within the healthcare sector \cite{}.
%The Cost of Sequencing a Human Genome, NIH
%GAVIN: Gene-Aware Variant INterpretation for medical sequencing, van der Velde 
From the collected data by genome sequencing experiments most of the analysis is handled in-silico due to the quantities of data that is produced. Proteins often find their use in diagnosing diseases experimentally \cite{}, however in-silico it is often limited to information about conservation in the amino acid sequence which may lead to identical results as by analyzing DNA.
% Introduction: Advances in Protein Analysis for the Clinical Laboratory, Hortin
%Protein Conformation and Diagnostic Tests: The Prion Protein, Bennion
Yet the 3D structure defines how a protein functions \cite{} and by assessing structures of protein variants it becomes possible to determine the change in function and diagnose protein variants that were unclassifiable through finding conservation.
%How do crystal lattice contacts reveal protein crystallization mechanism?, Nanev
Another advantage of the structural information is that it gives the possibility to develop treatment for diseases that are caused by a mutations. 
With experimental methods such as X-ray crystallography and nuclear magnetic resonance (NMR) more than 158000 structures \cite{} have been completely revealed, however it is only a tiny fraction of the potential proteins possible without folds \cite{}.
% wwPDB: Deposition Statistics, wwPDB	
%Chemiotics: How many proteins can we make? : The Sceptical Chymist, Cantrill
Making 3D structures is currently not common for diagnosis because it is relative expensive and is difficult to perform, some structures contain flexible regions which makes it hard to determine their exact position and can cause information loss about the structure \cite{}.
%PDB101: Learn: Guide to Understanding PDB Data: Missing Coordinates and Biological Assemblies, PDB101
%From Protein Structure to Function With Bioinformatics page 4 (book) 18 (pdf), Ridgen
\label{subsec:GD_Addition_of_structural_data}
\newpage

\subsection{Protein modeling techniques}
An alternative approach to determine structures is based on modeling the protein structure computationally from the amino acid sequence of the desired protein. A downside from computer generated models is that they do not follow the rules of physics and therefore not automatically fold into the correct confirmation. With the method homology modeling sequences of the requested protein are aligned to sequences of known experimental determined structures, based on these alignments a template is formed whereon structural fragments are built, it is not recommended to use this strategy if the sequence identity is less than 20\% since there might not be any structural relation at that point \cite{}.
% The relation between the divergence of sequence and structure in proteins, Chotia and Lesk
Another approach is protein threading which relies on the observation of folds in previous determined experimental structures. Based on the occurrence of specific folds a probability is predicted that a certain residue in a protein might fold in that manner. 

Strategies are continuously being improved and developed for proteins to determine the unknown structures, but all have the similar guidelines in avoiding steric hindrance \cite{} and finding the lowest energies based on different scoring systems \cite{}. From the computer generated models many are less accurate than the experimental determined methods and are often compared to them for reference. However the computational models do not have follow the same laws of physics which bottleneck the current experimental methods in for example determining membrane proteins \cite{}.
%X-ray crystallography at the heart of life science, Yonath
% Ramachandran plot, Wikipedia
%Scoring Tutorial , Shourya et al.
\label{subsec:GD_Protein_modeling_techniques}

\subsection{A theoretical large scale implementation of structural protein variant assessment}
With the wide spectrum of potential different proteins it can be difficult and maybe momentarily impossible to produce any form of universal protein assessment standardization that is able to determine if a mutation is harmful or not based on structural information. However a first step to solve such a complex problem would be by determining the correct approach, in this case it is assumed that a machine learning approach would be the best method for detecting patterns in structures and classifying the effect of structural changes. Because it has the ability to learn from structural mutations currently available, assuming that the current knowledge about structures and mutations is correct, and is able to develop new insights in how structural changes could affect proteins. 

Since the problem is so complex it should be divided into smaller more feasible problems, beginning by separating the different protein classes, which  for example can be done according to The Structural Classification of Proteins database (SCOP) \cite{}. A first discrimination between the proteins could be made based on protein type/fold class (membrane, globular, fibrous and disordered -proteins) because these differences already predetermine some functions and locations for certain proteins in a cell \cite{}. After formation of these classes each should have its own machine learning method applied so their features can be analyzed within context of where and how they function. The next set of discriminators is highly dependent on the variations in classes, but all have features in the end describing bonds, interactions and movement of complexes in protein structures. When for each of the main classes a method has been developed a meta classifier will determine based on certain aspects which method should be applied to determine the effect of mutation in a protein.
%SCOP2 prototype: a new approach to protein structure mining, Andreeva et. al.
%Membrane protein, Wikipedia
%Globular protein, Wikipedia
%Scleroprotein, Wikipedia
%Intrinsically disordered proteins, Wikipedia


%Within DNA sequences of humans continuously mutations occur from internal and external factors, some of these mutation insert, remove or substitute nucleotides which alter the the rate of tr
% Some types of mutations can be repaired \cite{}, other mutations are made on purpose\cite{}, incorrectly repaired \cite{} or are irreparable by the available mechanisms \cite{} and will be carried over to the next generation of cells or even to the offspring.
%DNA repair, Wikipedia
%Janewayâ€™s Immunobiology page 410-411 10.7: Germinal center B cells undergo V-region somatic hypermutation, and cells with mutations that improve af nity for antigen are selected, Kenneth murphy
% Repair of naturally occurring mismatches can induce mutations in flanking DNA, Chen et al.
% Error-Prone Repair of DNA Double-Strand Breaks, Rodgers and McVey


%
%Evidence is based on conservation whether it works or not
%
%Telling about the current state of gene variant prediction in the perspective of machine learning, GAVIN, SIFT, CADD

